ngraph_bridge/enable_variable_ops/ngraph_catalog.cc

```diff
@@ -68,89 +92,114 @@ bool NGraphCatalog::EncapOutputIndexNeedsCopy(int graphid, string node_name,
+const bool NGraphCatalog::ExistsInInputVariableSharedNameMap(
+    const string& key) {
   auto itr = NGraphCatalog::input_variable_sharedname_map_.find(key);
   return itr != NGraphCatalog::input_variable_sharedname_map_.end();
 }
 
-bool NGraphCatalog::ExistsInInputVariableSharedNameMap(int graphid,
-                                                       string node_name,
-                                                       int input_index) {
+const bool NGraphCatalog::ExistsInInputVariableSharedNameMap(
+    const int& graphid, const string& node_name, const int& input_index) {
   return NGraphCatalog::ExistsInInputVariableSharedNameMap(
       NGraphCatalog::CreateNodeKey(graphid, node_name, input_index));
 }
 
-void NGraphCatalog::DeleteFromInputVariableSharedNameMap(string key) {
+void NGraphCatalog::DeleteFromInputVariableSharedNameMap(const string& key) {
   NGraphCatalog::input_variable_sharedname_map_.erase(key);
 }
 
 // Functions for EncapOutputInfo Map
-void NGraphCatalog::AddToEncapOutputInfoMap(string key,
-                                            tuple<string, bool, bool> val) {
-  NGraphCatalog::encap_output_info_map_[key] = val;
+void NGraphCatalog::AddToEncapOutputInfoMap(
+    const string& key, const tuple<string, bool, bool>& val) {
+  if (NGraphCatalog::ExistsInEncapOutputInfoMap(key)) {
+    throw runtime_error(
+        "Trying to add an already existing key in EncapOutputInfo Map");
+  }
+  NGraphCatalog::encap_output_info_map_.insert({key, val});
 }
 
-void NGraphCatalog::AddToEncapOutputInfoMap(string key, string shared_name,
-                                            bool copy_to_tf,
-                                            bool is_tf_just_looking) {
+void NGraphCatalog::AddToEncapOutputInfoMap(const string& key,
+                                            const string& shared_name,
+                                            const bool& copy_to_tf,
+                                            const bool& is_tf_just_looking) {
+  if (NGraphCatalog::ExistsInEncapOutputInfoMap(key)) {
+    throw runtime_error(
+        "Trying to add an already existing key in EncapOutputInfo Map");
+  }
+
   // create a tuple
   tuple<string, bool, bool> val =
       make_tuple(shared_name, copy_to_tf, is_tf_just_looking);
-  NGraphCatalog::encap_output_info_map_[key] = val;
+  NGraphCatalog::encap_output_info_map_.insert({key, val});
 }
 
-bool NGraphCatalog::ExistsInEncapOutputInfoMap(string key) {
+const bool NGraphCatalog::ExistsInEncapOutputInfoMap(const string& key) {
   auto itr = NGraphCatalog::encap_output_info_map_.find(key);
   return itr != NGraphCatalog::encap_output_info_map_.end();
 }
 
-bool NGraphCatalog::ExistsInEncapOutputInfoMap(int graphid, string node_name,
-                                               int input_index) {
-  std::string key =
-      NGraphCatalog::CreateNodeKey(graphid, node_name, input_index);
+const bool NGraphCatalog::ExistsInEncapOutputInfoMap(const int& graphid,
```

Comment:
`const bool`.. perhaps you meant `const bool&`?
Also for bools and ints, just returning `bool` vs returning `const bool&` wont make much of a difference.

Similarly the return type of `CreateNodeKey` is `const string`. maybe you meant `const string&`?

Reply:
It is const string/bool and not & because the value being returned is created inside the function. returning const coz dont want the user of the api to change its value.

Reply:
Will change it to return string/bool as the user might still read them into non-const types and modify it